ksql-rest-app/src/main/java/io/confluent/ksql/rest/util/KsqlInternalTopicUtils.java

```diff
@@ -97,11 +97,14 @@ public static void ensureTopic(final String name,
       final ImmutableMap<String, Object> requiredConfig =
           ImmutableMap.of(TopicConfig.RETENTION_MS_CONFIG, requiredTopicRetention);
 
-      if (topicClient.addTopicConfig(name, requiredConfig)) {
-        log.info(
-            "Corrected retention.ms on ksql internal topic. topic:{}, retention.ms:{}",
-            name,
-            requiredTopicRetention);
+      if (!topicClient.getTopicConfig(name).getOrDefault(TopicConfig.RETENTION_MS_CONFIG, "0")
```

Comment:
doesn't `addTopicConfig` already do this internally?

Reply:
Ah you're right
https://github.com/confluentinc/ksql/blob/master/ksql-engine/src/main/java/io/confluent/ksql/services/KafkaTopicClientImpl.java#L217